code/datums/components/radioactive.dm

```diff
@@ -37,11 +37,15 @@
 	return ..()
 
 /datum/component/radioactive/process()
-	if(hl3_release_date && prob(50))
-		radiation_pulse(parent, strength, RAD_DISTANCE_COEFFICIENT*2, FALSE, can_contaminate)
-		strength -= strength / hl3_release_date
-		if(strength <= RAD_BACKGROUND_RADIATION)
-			qdel(src)
+	if(!prob(50))
+		return
+	radiation_pulse(parent, strength, RAD_DISTANCE_COEFFICIENT*2, FALSE, can_contaminate)
+
+	if(!hl3_release_date)
```

Comment:
eh

Reply:
the whole contamination things started as an excuse to add a half life reference